core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -23,15 +23,30 @@
 import java.util.Map;
 import org.apache.iceberg.ManifestReader.FileType;
 import org.apache.iceberg.exceptions.RuntimeIOException;
+import org.apache.iceberg.io.CloseableIterable;
 import org.apache.iceberg.io.FileIO;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.io.OutputFile;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
 
 public class ManifestFiles {
   private ManifestFiles() {
   }
 
+  /**
+   * Returns a {@link CloseableIterable} of file paths in the {@link ManifestFile}.
+   *
+   * @param manifest a ManifestFile
+   * @param io a FileIO
+   * @return a manifest reader
+   */
+  public static CloseableIterable<String> readPaths(ManifestFile manifest, FileIO io) {
```

Comment:
Do we want to document or reflect in the name that we return paths for liveEntries only?

Reply:
No, because the live entries are the expected case in the API. That's why readers are Iterable<DataFile>. Entries and deleted files are hidden through the API.